### PR TITLE
Add GraphQL query support for direct connections

### PR DESCRIPTION
**Pull request by rhauch**
## Summary of Changes

The graphQL functionality for reading cluster resources was not quite working for direct connections. This adds that support and adds some unit tests to verify the queries function.

Limitations
* Cluster information returned in the GraphQL query results do not yet reflect whether the user has connectivity with or the ability to authenticate with the Kafka and Schema Registry clusters referenced in direct connections. Connection status will be added soon.
* Kafka REST and SR APIs also are not yet working with direct connections, and this will be fixed soon.

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

- Tests:
    - [x] Added new
    - [x] Updated existing
    - [ ] Deleted existing
- [x] Have you validated this change locally against a running instance of the Quarkus dev server?
    ```shell
    make quarkus-dev
    ```
- [x] Have you validated this change against a locally running native executable?
    ```shell
    make mvn-package-native && ./target/ide-sidecar-*-runner
    ```

### CODE REVIEW

**Review by rohitsanj — APPROVED** [review]
LGTM

File: src/test/java/io/confluent/idesidecar/restapi/util/SchemaRegistryContainer.java [hunk 1]
```diff
@@ -23,4 +24,8 @@ public SchemaRegistryContainer(String dockerImageName, String kafkaBootstrap) {
   public String endpoint() {
     return String.format("http://%s:%d", getHost(), getMappedPort(PORT));
   }
+
+  public String getClusterId() {
```
[the remark this answers is not in the document]
Comment (rhauch) [reply src/test/java/io/confluent/idesidecar/restapi/util/SchemaRegistryContainer.java:28]: Yeah, that slipped in from #134. I could remove it, but I'm not sure it's worth the effort.

File: src/test/java/io/confluent/idesidecar/restapi/resources/DirectQueryResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        "graph/real/direct-connections-no-schema-registry-expected.json"
+    );
+  }
+
+  @Test
+  void shouldGetDirectConnectionsWithOnlySchemaRegistry() {
+    // When there is a direct connection
+    createDirectConnection(false, true);
+
+    // Then the response should contain the connection and its Kafka cluster but no Schema Registry
```
[the remark this answers is not in the document]
Comment (rhauch) [reply src/test/java/io/confluent/idesidecar/restapi/resources/DirectQueryResourceTest.java:68]: Fixed.

File: src/test/java/io/confluent/idesidecar/restapi/util/SchemaRegistryContainer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rohitsanj) [reply src/test/java/io/confluent/idesidecar/restapi/util/SchemaRegistryContainer.java:28]: All good

**Review by rohitsanj — APPROVED** [review]

File: src/generated/resources/schema.graphql [hunk 3]
```diff
@@ -59,6 +59,27 @@ type ConfluentLocalKafkaCluster {
...
+  name: String!
+  schemaRegistry: DirectSchemaRegistry
+  type: ConnectionType!
+}
+
+type DirectKafkaCluster {
+  bootstrapServers: String!
+  connectionId: String!
+  id: String!
+  uri: String
```
Comment (flippingbits) [inline src/generated/resources/schema.graphql:74]: Out of curiosity, what does `uri` hold in the case of a `DirectKafkaCluster`?

Comment (rhauch) [reply src/generated/resources/schema.graphql:74]: Good question. First, in the GraphQL schema, the `Cluster` interface defines `uri` and `DirectKafkaCluster` inherits the field whether or not it needs it.

Second, I think that's okay, because the `CCloudKafkaCluster` and `ConfluentLocalKafkaCluster` subtypes expose the Kafka REST proxy URL via that `uri` field, and we will want the `DirectKafkaCluster` to be able to expose the Kafka REST proxy URL if there is one and we know about it. I do anticipate in the near future enhancing the `kafka_cluster` config in direction connections to optionally include the REST endpoint, either purely as informational, or to be able to specify the REST endpoint instead of `bootstrapServers` so we can then look up the latter (and more Kafka cluster metadata) from the former like we currently do in the local connections.

Bottom line: it is currently a bit weird now, but I think we'll make it work in the next sprint.

**Comment by rhauch** [comment]
This was based on #131, and since that was just merged this PR's base branch changed and invalidated your review/approvals. 

@flippingbits, @rohitsanj: mind taking another pass? There are no other changes since your previous reviews.

**Comment by rhauch** [comment]
Fixed the merge conflicts with `main`, due to #135.

**Review by rohitsanj — APPROVED** [review]

File: src/generated/resources/schema.graphql (hunk 3 above)
Comment (flippingbits) [reply src/generated/resources/schema.graphql:74]: Thanks for the clarification! I didn't know we want to support interacting with `DirectKafkaCluster`s via the REST proxy. Also, as you said, the [KafkaClusterConfig doesn't seem to offer the uri field](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/java/io/confluent/idesidecar/restapi/models/ConnectionSpec.java#L246) right now so I'm wondering how/if we'd fill the `uri` field of the GraphQL API's `DirectKafkaCluster`. Let's get back to it in one of the next sprints per your suggestion.

Comment (rhauch) [reply src/generated/resources/schema.graphql:74]: > I didn't know we want to support interacting with DirectKafkaClusters via the REST proxy.
Just for bootstrapping to collect metadata like cluster ID. We still plan to use the internal REST proxy for functionality rather than rely upon the provided REST proxy.
